gammapy/stats/variability.py

```diff
@@ -228,3 +230,102 @@ def compute_flux_doubling(flux, flux_err, coords, axis=0):
     }
 
     return doubling_dict
+
+
+def TKAlgorithm(
+    power_spectrum,
+    npoints,
+    spacing,
+    type="discrete",
+    random_state="random-seed",
+    normalization=1.0,
+):
+    """Implementation of the Timmer-Koenig algorithm to simulate a time series from a power spectrum.
+
+    Parameters
+    ----------
+    power_spectrum : float, `~numpy.ndarray`
+        Power spectrum used to generate the time series. It can either be presented analytically, as a parameter or set
+        of parameters of a function, or as a `~numpy.ndarray` describing the spectrum in a discrete manner.
+    npoints : float
+        Number of points in the output time series.
+    spacing : float
+        Sample spacing, inverse of the sampling rate.
+    type : {"discrete", "powerlaw", "white"}
+        Type of input periodogram. For `~numpy.ndarray` inputs, the type is "discrete". Default is "discrete".
+        If 'type' = "discrete", 'powerspectrum' is expected to be  a `~numpy.ndarray` containing the discrete spectrum.
+        If 'type' = "powerlaw", 'powerspectrum' is expected to be  a float representing the power index.
+        If 'type' = "white", 'powerspectrum' is expected to be  a float representing the standard deviation.
+    random_state : {int, 'random-seed', 'global-rng', `~numpy.random.RandomState`}
+        Defines random number generator initialisation.
+        Passed to `~gammapy.utils.random.get_random_state`. Default is "random-seed".
+    normalization : float
+        Normalization factor to be applied to the final time series
+
+    Returns
+    -------
+    time_series: `~numpy.ndarray`
+        Simulated time series.
+
+    References
+    ----------
+    ..[Timmer1995]"On generating power law noise", J. Timmer and M, Konig, section 3.
+    """
+    if isinstance(power_spectrum, np.ndarray) and type != "discrete":
+        raise ValueError(
+            "Power spectrum input is an array but the requested type is not 'discrete'."
+        )
+
+    random_state = get_random_state(random_state)
+
+    frequencies = np.fft.fftfreq(npoints, spacing)
+
+    # To obtain real data only the positive or negative part of the frequency is necessary.
+    real_frequencies = np.sort(np.abs(frequencies[frequencies < 0]))
+
+    if type == "discrete":
+        periodogram = power_spectrum
```

Comment:
I think this is a bit dangerous as is. This is not tested and there is no way to make sure frequencies and power spectrum match.